src/main/java/com/github/yusufyilmazfr/podcast4j/util/PodcastDeserializer.java

```diff
@@ -0,0 +1,24 @@
+package com.github.yusufyilmazfr.podcast4j.util;
+
+import com.fasterxml.jackson.core.JacksonException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.github.yusufyilmazfr.podcast4j.entity.Podcast;
+
+import java.io.IOException;
+
+public class PodcastDeserializer extends JsonDeserializer<Podcast> {
+
+    @Override
+    public Podcast deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException, JacksonException {
+        JsonNode node = jsonParser.getCodec().readTree(jsonParser);
+
+        // handle empty array when no feed
+        if (node.isArray() && node.isEmpty()) {
```

Comment:
Awesome work! I liked it. :) I remember the problem, response returns a object, but if there is no any object response is an empty array. Is it true, am i missing something @konieshadow?

Reply:
Just run  the `getLiveEpisodes_shouldReturnMatchedEpisodes()` unit test. This call should got response content as bellow. The `categories` field of second item is '[]'.

```JSON
{
    "status": "true",
    "items": [
        {
            "id": 987040,
            "title": "The Podcast - A Conversation with Barry Luijbregts",
            "link": "https:\/\/serve.podhome.fm\/episodepage\/the-podcast-with-joe\/1",
            "guid": "5fd91f95-d80d-49d1-af0d-85eb3408a398",
            "datePublished": 1732494593,
            "datePublishedPretty": "November 24, 2024 6:29pm",
            "dateCrawled": 1732494147,
            "enclosureUrl": "http:\/\/stream.podhome.fm\/the-podcast",
            "enclosureType": "audio\/mpeg",
            "enclosureLength": 0,
            "explicit": 0,
            "image": "",
            "feedItunesId": null,
            "feedImage": "https:\/\/assets.podhome.fm\/c2d95029-4e8c-44d1-680b-08dc161e8125\/638660286268309371CoverArt%20PNG%203000px.jpg",
            "feedId": 7111253,
            "feedTitle": "The Podcast",
            "feedLanguage": "en-us",
            "categories": {
                "20": "Education",
                "55": "News",
                "54": "Commentary"
            }
        },
        {
            "id": 986859,
            "title": "Liberty Fellowship Live - Every Sunday at 2:30 PM (Mountain Time)",
            "link": "https:\/\/videohaven.com\/w\/ijed1qg3e9MPDyKRTj41Hh",
            "guid": "8c36e25a-a4f4-4906-9e6b-2520dcc54d92_2024-11-24T20:18:10.523Z",
            "datePublished": 1732480453,
            "datePublishedPretty": "November 24, 2024 2:34pm",
            "dateCrawled": 1732480453,
            "enclosureUrl": "https:\/\/videohaven.com\/static\/streaming-playlists\/hls\/8c36e25a-a4f4-4906-9e6b-2520dcc54d92\/master.m3u8",
            "enclosureType": "application\/x-mpegURL",
            "enclosureLength": 0,
            "explicit": 0,
            "image": "",
            "feedItunesId": null,
            "feedImage": "https:\/\/videohaven.com\/lazy-static\/avatars\/454b6d6d-64fb-4f30-aea0-a0bf116406c3.jpg",
            "feedId": 6588245,
            "feedTitle": "Liberty Fellowship",
            "feedLanguage": "",
            "categories": []
        },
        {
            "id": 986691,
            "title": "Womens Hour with Co Host Jo Wood",
            "link": "https:\/\/serve.podhome.fm\/episodepage\/womens-hour\/26",
            "guid": "cc2185a1-271c-4cbb-8f0e-75ab5f9bc73b",
            "datePublished": 1732522508,
            "datePublishedPretty": "November 25, 2024 2:15am",
            "dateCrawled": 1732469307,
            "enclosureUrl": "http:\/\/stream.podhome.fm\/womens-hour",
            "enclosureType": "audio\/mpeg",
            "enclosureLength": 0,
            "explicit": 0,
            "image": "",
            "feedItunesId": 1754628873,
            "feedImage": "https:\/\/assets.podhome.fm\/09594611-fcf8-4722-f899-08dc87bedc7d\/638559778271921666DSC_7455aajpg.jpg",
            "feedId": 6939020,
            "feedTitle": "Women's Hour",
            "feedLanguage": "en-gb",
            "categories": {
                "77": "Society",
                "78": "Culture",
                "29": "Health",
                "30": "Fitness"
            }
        },
        {
            "id": 986676,
            "title": "No Agenda Episode 1715 - Live",
            "link": "http:\/\/trollroom.io",
            "guid": "NA-1715 Live",
            "datePublished": 1732529208,
            "datePublishedPretty": "November 25, 2024 4:06am",
            "dateCrawled": 1732468074,
            "enclosureUrl": "https:\/\/op3.dev\/e\/listen.noagendastream.com\/noagenda",
            "enclosureType": "audio\/mpeg",
            "enclosureLength": 33,
            "explicit": 0,
            "image": "",
            "feedItunesId": 269169796,
            "feedImage": "https:\/\/noagendaassets.com\/enc\/1732488838.618_na-1715-art-feed.jpg",
            "feedId": 41504,
            "feedTitle": "No Agenda",
            "feedLanguage": "en",
            "categories": {
                "55": "News",
                "59": "Politics",
                "16": "Comedy"
            }
        },
        {
            "id": 986521,
            "title": "Supporting Your Kids Education with Carla Hochstetler",
            "link": "https:\/\/dudesanddadspodcast.com\/live",
            "guid": "5db53682-1908-4431-b431-0d74ad96ad99",
            "datePublished": 1732529042,
            "datePublishedPretty": "November 25, 2024 4:04am",
            "dateCrawled": 1732451415,
            "enclosureUrl": "https:\/\/mira.streamerr.co\/listen\/lehmancreations\/radio.mp3",
            "enclosureType": "audio\/mpeg",
            "enclosureLength": 312,
            "explicit": 0,
            "image": "",
            "feedItunesId": 1455296843,
            "feedImage": "https:\/\/feeds.podcastmirror.com\/~images\/2200361732502769.jpeg",
            "feedId": 955598,
            "feedTitle": "Dudes And Dads Podcast",
            "feedLanguage": "en-US",
            "categories": {
                "77": "Society",
                "78": "Culture",
                "85": "Relationships",
                "36": "Kids",
                "37": "Family",
                "38": "Parenting"
            }
        }
    ],
    "count": 5,
    "max": "5",
    "description": "Found matching items."
}
```